### PR TITLE
INT-743 fix for ampersand-state@4.8.0

### PR DESCRIPTION
**Pull request by rueckstiess**

### CODE REVIEW

File: models/leafclause.js [hunk 1]
```diff
@@ -77,8 +77,8 @@ module.exports = Clause.extend({
   },
   initialize: function(attrs, options) {
     // initialize value manually (since it is polymorphic)
-    if (attrs) {
-      this.value = this._initializeValue(attrs.value, options);
+    if (this.value !== undefined) {
+      this.value = this._initializeValue(this.value, options);
```
Comment (kangas) [inline models/leafclause.js]: This is the source of incompatibility with 4.7.0+? Definitely a subtle detail. Also interesting that null worked previously.

**Comment by WaleyChen** [comment]
LGTM! Null bug is gone. :smile:

**Comment by kangas** [comment]
When trying to integrate this into compass (and removing the version pin on ampersand-state), my connection dialog loads blank and I see the following error in the console log.

>  "stacktrace": "TypeError: Invalid data type of `function` for `chart` property. Use one of the default types or define your own\n    at createPropertyDefinition (file:///Users/<redacted>/workspace/compass/dist/MongoDB%20Compass-darwin-x64/MongoDB%20Compass.app/Contents/Resources/app/index.js:3300:15)\n    at file:///Users/<redacted>/workspace/compass/dist/MongoDB%20Compass-darwin-x64/MongoDB%20Compass.app/Contents/Resources/app/index.js:3512:21\n    at file:///Users/<redacted>/workspace/compass/dist/MongoDB%20Compass-darwin-x64/MongoDB%20Compass.app/Contents/Resources/app/index.js:48125:11\n    at baseForOwn (file:///Users/<redacted>/workspace/compass/dist/MongoDB%20Compass-darwin-x64/MongoDB%20Compass.app/Contents/Resources/app/index.js:52388:10)\n    at file:///Users/<redacted>/workspace/compass/dist/MongoDB%20Compass-darwin-x64/MongoDB%20Compass.app/Contents/Resources/app/index.js:52403:12\n    at Function.extend (file:///Users/<redacted>/workspace/compass/dist/MongoDB%20Compass-darwin-x64/MongoDB%20Compass.app/Contents/Resources/app/index.js:3511:17)\n    at Object.479../svg-template.jade (file:///Users/<redacted>/workspace/compass/dist/MongoDB%20Compass-darwin-x64/MongoDB%20Compass.app/Contents/Resources/app/index.js:97256:29)\n    at s (file:///Users/<redacted>/workspace/compass/dist/MongoDB%20Compass-darwin-x64/MongoDB%20Compass.app/Contents/Resources/app/index.js:1:254)\n    at file:///Users/<redacted>/workspace/compass/dist/MongoDB%20Compass-darwin-x64/MongoDB%20Compass.app/Contents/Resources/app/index.js:1:305\n    at Object.473../array-root (file:///Users/<redacted>/workspace/compass/dist/MongoDB%20Compass-darwin-x64/MongoDB%20Compass.app/Contents/Resources/app/index.js:96393:15)",

Alas, since our source maps aren't working, this is a nuisance to debug. :(

**Comment by rueckstiess** [comment]
This looks like an unrelated issue in Compass. Don't see how the chart property is affected by the language-model. I'll merge this now and publish a new version, and then diagnose from the Compass side.

**Pull request merged**